src/webgpu/shader/validation/extension/dual_source_blending.spec.ts

```diff
@@ -317,3 +317,246 @@ ${kStageIOValidationTests[t.params.attr].shader}
 `;
     t.expectCompileResult(kStageIOValidationTests[t.params.attr].pass, code);
   });
+
+const kUsageValidationTests = {
+  const: {
+    code: `@blend_src(0) const color = 1.2;`,
+    pass: false,
+    use_default_main_function: true,
+  },
+  override: {
+    code: `@blend_src(0) @id(0) override color : vec4f;`,
+    pass: false,
+    use_default_main_function: true,
+  },
+  let: {
+    code: `
+    @fragment fn main() -> vec4f {
+      let @blend_src(0) color = vec4f();
+      return color;
+    }
+    `,
+    pass: false,
+    use_default_main_function: false,
+  },
+  var_private: {
+    code: `@blend_src(0) var<private> color : vec4f;`,
+    pass: false,
+    use_default_main_function: true,
+  },
+  var_function: {
+    code: `
+    @fragment fn main() -> vec4f {
+      var @blend_src(0) color : vec4f;
+      color = vec4f();
+      return color;
+    }
+    `,
+    pass: false,
+    use_default_main_function: false,
+  },
+  function_declaration: {
+    code: `
+    @blend_src(0) fn fun() {}
+    `,
+    pass: false,
+    use_default_main_function: true,
+  },
+  non_entrypoint_function_input_non_struct: {
+    code: `
+    fn fun(@blend_src(0) color : vec4f) -> vec4f {
+      return color;
+    }
+    `,
+    pass: false,
+    use_default_main_function: true,
+  },
+  non_entrypoint_function_output_non_struct: {
+    code: `
+    fn fun() -> @blend_src(0) vec4f {
+      return vec4f();
+    }
+    `,
+    pass: false,
+    use_default_main_function: true,
+  },
+  entrypoint_input_non_struct: {
+    code: `
+    @fragment fn main(@location(0) @blend_src(0) color : vec4f) -> vec4f {
```

Comment:
Suggest adding a `@location(0)` to the return type, otherwise this test may be failing for the wrong reason.